### ISSUE TITLE
Why not have the NSwag studio as an extension to VS

### ISSUE THREAD

**Issue by weitzhandler** [NSwag/NSwag#548]
It should be a lot easier if it's part of Visual Studio as a VSIX.
The entire generation process could be automated easily.

**Comment by rsuter** [comment]
There are currently no plans for a Visual Studio extension. First it is a big pain to build VS extensions (compared to a simple WPF UI) and second I just don't have the time to also support a VS extension without being paid for it... :-)

**Comment by weitzhandler** [comment]
An extension is plain WPF...
But I feel you! And thanks for your time and effort for everything you achieved so far it's great!

**Comment by rsuter** [comment]
True, but if we build a VS extension then it should bring more feature than just the standalone NSwagStudio application. For example it should integrate in the project system, so that you can add NSwag to an existing Web API project, or provide context menu on the client project to regenerate the code... maybe we also need project templates for better and simpler integration.

**Comment by rsuter** [comment]
Maybe an extension with the binary and context menu to create and run an nswag.json file?

**Comment by weitzhandler** [comment]
Yes. That would be a great starting point indeed.
